packages/core/src/Synchronizer.ts

```diff
@@ -515,6 +518,22 @@ export class Synchronizer extends EventEmitter {
         return tree
     }
 
+    async genHistoryTree(): Promise<IncrementalMerkleTree> {
+        const tree = new IncrementalMerkleTree(this.settings.historyTreeDepth)
+        const leaves = await this._db.findMany('HistoryTreeLeaf', {
+            where: {
+                attesterId: this.attesterId.toString(),
+            },
```

Comment:
I prefer that merge #367 before this 
and set `attesterId` as its optional input here

some bugs are detected in #367 that the epoch key index is not universal among each attester
[45a94fa](https://github.com/Unirep/Unirep/pull/367/commits/45a94faa0f29c0fc8ffad1a0dc4afc51bd025b12)

Reply:
> some bugs are detected in https://github.com/Unirep/Unirep/pull/367 that the epoch key index is not universal among each attester

can you describe this more?